rebar.config

```diff
@@ -1 +1,6 @@
 {plugins, [rebar3_hex]}.
+
+{pre_hooks, [
+    {compile, "cargo build --release"},
+    {compile, "sh -c \"cp $(cat so-path) priv/libnative.so\""}
```

Comment:
what's the point of doing the match on `dylib` and `dll` compilation targets if this one's going to hard-code the target file as `.so`?

Reply:
hmm, that is a good point, cc: @JayKickliter

Reply:
If you're talking about the [match] in `build.rs`:

We can't specify the library extension. It's determined by the platform. Even more so, we can't specify where the compiled library, regardless of extensions, winds up. It depends on build type, whether or not we're cross-compiling, platform. So the entire logic of `build.rs` is so we can know  (in Rebar) the full path to the compiled library by writing it into file named `so-path`. Lastly, erlang insists on NIFs having the `.so` extension.

[match]: https://github.com/helium/vincenty/pull/2/files#diff-358ae2245aa2cd99768b0c568a07648d4c26c8cdd84b15cd6f0af2680ec85240R18-R22

Reply:
Please let me know if this doesn't make sense. I know this question will come up again, and pretty soon I won't remember the rationale.

Reply:
Well atleast now we will have the rationale here :slightly_smiling_face:

Reply:
I think it makes sense if perhaps some of the naming conventions are confusing. For me, the CARGO_CFG_TARGET_OS envar implies I can set the target platform to compile for, i.e. build a `.dll` for a windows machine from my Linux laptop. If Erlang is cool with that and just cares about seeing the right file extension that makes total sense to me why we do what we do in the rebar pre-hook. Conversely, if we’re only ever targeting running on Linux hardware, VMs or Linux-based Docker containers why not just always target a `.so` and fail fast if a user mistakenly tries to do otherwise? Are you saying we can’t target a platform to compile for and that environment variable is just pulling from the local build environment? (meaning our CI server would always be setting the production release targets to a Linux system)

Reply:
We don’t set those ENV vars. They’re defined and set by the rust build tool. And we target linux, macOS, FreeBSD. I also don’t want to leave out windows, but I don’t know if this would build there

Reply:
One thing is for sure: we use this hack all over the place so I need to document its purpose